File: leetcode/easy/move-zeroes/move-zeroes.js
/*
Given an integer array nums, move all 0's to the end of it while maintaining the relative order of the non-zero elements.
Note that you must do this in-place without making a copy of the array.
*/

/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {

    for (let i = nums.length - 2; i >= 0; i--) {
        if (nums[i] === 0) {
            nums.push(nums.splice(i, 1))
        }
    }
};

File: leetcode/easy/move-zeroes/move-zeroes.py
'''
Given an integer array nums, move all 0's to the end of it while maintaining the relative order of the non-zero elements.
Note that you must do this in-place without making a copy of the array.
'''

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        for num in nums:
            if num == 0:
                nums.append(nums.pop(nums.index(num)))